app/segments/drill/[id]/submission/input.js

```diff
@@ -526,6 +526,9 @@ export default function Input({ setToggleResult, setOutputData }) {
   const [invalidDialogVisible, setInvalidDialogVisible] = useState(false);
   const hideInvalidDialog = () => setInvalidDialogVisible(false);
 
+  const [dialogVisible, setDialogVisible] = useState(false);
```

Comment:
On line 200ish of this file (uploadNewLeaderboard function), there are some remaining `alert(e)`, should we try to use the dialogmessage component? I wasn't sure if it would get complicated passing the error message from within a function to the "main function" (line 781)

Reply:
Hopefully I did the rebase right. For further testing on the input, let's merge https://github.com/Golf-Drill-Challenge-App/Golf-App/pull/218 (numshot bug PR) first and rebase this branch. Too many bugs when trying to test drill submission on this branch.

Reply:
I think the "cleanest" solution would be to put all of the functions inside the functional component. I would rather not pass these states into them.

Reply:
I think I just ended up doing a `throw e` from the inner try-catch blocks, and then doing the setDialogMessage code inside the outermost catch block (in the "main function" at line 700 ish of input.js), which mostly avoided actually passing React state I think. Not sure if that's what you meant.

Reply:
Frank's feedback: Maybe integrate showDialog function into other files, and combine with current if-else logic for dialog messages.

However, maybe move setDialogMessage into the root component to make react state handling simpler

Reply:
Also. need to remove nested try catches, maybe just one try catch around the parent function